File: src/CSharp/788.RotatedDigits.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace PeefyLeetCode.RotatedDigits
{
public class Solution {
    int[] change = {0 , 0 , 1 , -1 , -1 , 1 , 1 , -1 , 0 , 1};
    
    bool ifGood(int n){
        bool result = false;
        while(n != 0){
            if(change[n%10] == -1)
                return false;
            else if(change[n%10] == 1){
                result = true;
            }
            n /= 10;
        }
        return result;
    }
    public int RotatedDigits(int N) {
        int cnt = 0;
        for(int i = 1 ; i <= N ; ++i){
           if(ifGood(i))
               ++cnt;
        }
        return cnt;
    }
}
}

File: src/Java/788.RotatedDigits.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;

class Solution {
    public int rotatedDigits(int N) {
        int res = 0;
        for (int num = 1;num <= N;++num){
            String numstr = String.valueOf(num);
            if ((numstr.contains("3") == false && numstr.contains("4") == false && numstr.contains("7") == false) && 
            (numstr.contains("2") == true || numstr.contains("5") == true || numstr.contains("6") == true || numstr.contains("9") == true)) {
                res++;
            }
        }
        return res;
    }
}

